pkg/app/pipedv1/plugin/kubernetes/livestate/plugin.go

```diff
@@ -75,15 +77,78 @@ func (p Plugin) GetLivestate(ctx context.Context, _ sdk.ConfigNone, deployTarget
 		resourceStates = append(resourceStates, m.ToResourceState(deployTarget.Name))
 	}
 
+	manifests, err := p.loadManifests(ctx, input, cfg.Spec, provider.NewLoader(toolRegistry))
+	if err != nil {
+		input.Logger.Error("Failed to load manifests", zap.Error(err))
+		return nil, err
+	}
+
+	liveManifests := make([]provider.Manifest, 0, len(namespacedLiveResources)+len(clusterScopedLiveResources))
+	liveManifests = append(liveManifests, namespacedLiveResources...)
+	liveManifests = append(liveManifests, clusterScopedLiveResources...)
+
+	// Calculate SyncState by comparing live manifests with desired manifests
+	// TODO: Implement drift detection ignore configs
+	diffResult, err := provider.DiffList(liveManifests, manifests, input.Logger,
+		diff.WithEquateEmpty(),
+		diff.WithIgnoreAddingMapKeys(),
+		diff.WithCompareNumberAndNumericString(),
+	)
+	if err != nil {
+		input.Logger.Error("Failed to calculate diff", zap.Error(err))
+		return nil, err
+	}
+
+	syncState := calculateSyncState(diffResult, input.Request.DeploymentSource.CommitHash)
+
 	return &sdk.GetLivestateResponse{
 		LiveState: sdk.ApplicationLiveState{
 			Resources:    resourceStates,
 			HealthStatus: sdk.ApplicationHealthStateUnknown, // TODO: Implement health status calculation
 		},
-		SyncState: sdk.ApplicationSyncState{}, // TODO: Implement sync state calculation
+		SyncState: syncState,
 	}, nil
 }
 
+func calculateSyncState(diffResult *provider.DiffListResult, commit string) sdk.ApplicationSyncState {
```

Comment:
reference implementation is here
https://github.com/pipe-cd/pipecd/blob/57c48c11ab89ad41ffc3ae140ce7661640bc4663/pkg/app/piped/driftdetector/kubernetes/detector.go#L385-L422